transport/tchannel/outbound.go

```diff
@@ -84,19 +84,24 @@ func (o *Outbound) Call(ctx context.Context, req *transport.Request) (*transport
 	if _, ok := ctx.(tchannel.ContextWithHeaders); ok {
 		return nil, errDoNotUseContextWithHeaders
 	}
-	root := o.transport.ch.RootPeers()
 	p, onFinish, err := o.getPeerForRequest(ctx, req)
 	if err != nil {
 		return nil, toYARPCError(req, err)
 	}
-	tp := root.GetOrAdd(p.HostPort())
-	res, err := o.callWithPeer(ctx, req, tp)
+	res, err := p.Call(ctx, req)
 	onFinish(err)
 	return res, toYARPCError(req, err)
 }
 
+// Call sends an RPC to this specific peer.
+func (p *tchannelPeer) Call(ctx context.Context, req *transport.Request) (*transport.Response, error) {
+	root := p.transport.ch.RootPeers()
+	tp := root.GetOrAdd(p.HostPort())
+	return callWithPeer(ctx, req, tp)
```

Comment:
optimization for future: can we hold a reference to the `tchannel.Peer` when the peer is created? That way we avoid this map lookup for each RPC